packages/didcomm/src/modules/message-pickup/storage/InMemoryMessagePickupRepository.ts

```diff
@@ -17,10 +17,10 @@ interface InMemoryQueuedMessage extends QueuedMessage {
 
 @injectable()
 export class InMemoryMessagePickupRepository implements MessagePickupRepository {
-  private logger: Logger
+  private logger?: Logger
   private messages: InMemoryQueuedMessage[]
 
-  public constructor(@inject(InjectionSymbols.Logger) logger: Logger) {
+  public constructor(@inject(InjectionSymbols.Logger) logger?: Logger) {
```

Comment:
Why optional?

Reply:
If we don't have access to this, we should just pass the agent context to all the methods in this class I think

Reply:
Yeah, I didn't want to change all `MessagePickupRepository` interface methods only for this basic implementation. But I think resistance is futile: a good refactoring of all of this will be needed before 0.6.0 release!